product/ci/engine/new/executor/addon_executor.go

```diff
@@ -59,6 +61,9 @@ func ExecuteStepOnAddon(ctx context.Context, step *pb.UnitStep, tmpFilePath stri
 	if err != nil {
 		log.Errorw("Execute step RPC failed", "step_id", stepID,
 			"elapsed_time_ms", utils.TimeSince(st), zap.Error(err))
+		if strings.Contains(err.Error(), "Error while dialing: dial tcp") {
```

Comment:
nit: Move `Error while dialing: dial tcp` to a constant.

`const addonDisconnectErr = "Error while dialing: dial tcp"`

```suggestion
		if strings.Contains(err.Error(), addonDisconnectErr) {
```